### PR TITLE
[feature] add min and max at static methods

### PR DESCRIPTION
**Pull request by itsmuntadhar**
First, I would like to say thanks for the great work!

We've been looking into ULID to use instead of UUID/GUID in our projects and figured we could get rid of the dedicated "CreatedAt" column/prop that we usually have.

To make querying easier, we thought such methods would be great to have in the lib so we could do something like

```cs
var anHourAgo = DateTimeOffset.UtcNow.AddHours(-1);
var idGte = Ulid.MinAt(anHourAgo);
queryable = queryable.Where(x => x.Id >= idGte);
```

Was also looking into writing a couple more extensions, maybe in another PR.
Basically

```cs
DateTimeOffset.UtcNow.NewUlid();
DateTimeOffset.UtcNow.MinUlid();
DateTimeOffset.UtcNow.MaxUlid();
```

Thanks again

### CODE REVIEW

**Comment by RobThree** [comment]
Hi!

I'm not sure why I missed this PR and never replied. I'll look into this soon(-ish) and get back to you. Just wanted to let you know your PR is appreciated and I apologize for missing it.

**Comment by itsmuntadhar** [comment]
Things happen :)

Will look into solving the conflicts shortly

**Comment by RobThree** [comment]
I think it would be a good idea to also support "just" `DateTime`, even though `DateTimeOffset` is clearly the better choice. I just don't think they're mutually exclusive.
